### PR TITLE
Questions to be asked on test entry

### PR DESCRIPTION
**Pull request by benwarfield-usds**
WIP for CDCGov/prime-central#56.

Known current issues:

- [x] no "Add to Queue" button at the top (just cancel). This is probably the biggest pain to fix, actually.
- [x] no "No Symptoms" checkbox, which needs a little poking at to get the least bad approach using our component library
- [ ] checked options are all funny looking, thanks to some CSS argument between us and the CMS design system CSS.
- [x] "cancel" doesn't actually work, which does admittedly seem like a bit of a problem
- [x] answers look like they are being stored, but they actually are not
- [ ] various radio buttons that are supposed to be in a single row, are not.
- [x] dialog needs to appear in enqueue flow as well
- [x] prior test and symptom dates should be validated
- [x] prior test details should be disabled if the answer to "first test" is Yes
- [x] we should have a warning dialog for saving incomplete answers
- [x] queue entries should be correctly tagged if the answers have been provided
- [x] test submission should also have a warning dialog for incomplete answers
- [x] editing answers from the queue page should persist the updated answsers

### CODE REVIEW

File: frontend/src/app/testQueue/testQueueSelectors.js [hunk 1]
```diff
@@ -6,47 +6,39 @@ import { getPatients } from "../patients/patientSelectors";
...
 export const getDetailedPatientsInTestQueue = createSelector(
-  _getPatientIdsInTestQueue,
   getPatients,
-  (testQueuePatientIds, allPatients) => {
-    let patientsInQueue = [];
-    testQueuePatientIds.forEach((patientId) =>
-      patientsInQueue.push(allPatients[patientId])
-    );
-    return patientsInQueue;
+  _getPatientsInTestQueue,
```
Comment (sharmaneil) [inline frontend/src/app/testQueue/testQueueSelectors.js:35]: thanks for this refactor. This grew into a mess as queue requirements changed, and I never got around to cleaning it up.

File: frontend/src/app/testQueue/testQueueSelectors.js [hunk 2]
```diff
@@ -6,47 +6,39 @@ import { getPatients } from "../patients/patientSelectors";
...
-const _getPatientIdsInTestQueue = createSelector(
-  _getPatientsInTestQueue,
-  (patients) => {
-    return Object.keys(patients)
-      .sort((a, b) => patients[a].dateAdded - patients[b].dateAdded);
-  }
-);
-
-// returns a subset of patient fields, including the queue status, for *ALL* patients
+// returns a superset of patient fields, including the queue status, for *ALL* patients
```
Comment (sharmaneil) [inline frontend/src/app/testQueue/testQueueSelectors.js:9]: 🎉

File: frontend/src/app/testQueue/testQueueSelectors.js (hunk 1 above)
Comment (benwarfield-usds) [reply frontend/src/app/testQueue/testQueueSelectors.js:35]: It was only a _little_ ball of mud... 😁

**Comment by dmethvin-gov** [comment]
@benwarfield-usds and I have conspired to make progress on this. He's going to make a few more changes and then merge it even though there are some display issues from the dueling design systems. Then I'll figure out a fix for that. If it turns out we can'f fix by Tuesday it's still possible to user test, the checks just look wonky.

**Pull request merged**